library/std/src/panicking.rs

```diff
@@ -744,19 +745,22 @@ fn rust_panic_with_hook(
     if let Some(must_abort) = must_abort {
         match must_abort {
             panic_count::MustAbort::PanicInHook => {
-                // Don't try to print the message in this case
-                // - perhaps that is causing the recursive panics.
+                // Don't try to format the message in this case, perhaps that is causing the
+                // recursive panics. However if the message is just a string, no user-defined
+                // code is involved in printing it, so that is risk-free.
+                let msg_str = message.and_then(|m| m.as_str()).map(|m| [m]);
+                let message = msg_str.as_ref().map(|m| fmt::Arguments::new_const(m));
                 let panicinfo = PanicInfo::internal_constructor(
-                    None,     // no message
-                    location, // but we want to show the location!
+                    message.as_ref(),
+                    location,
                     can_unwind,
                     force_no_backtrace,
                 );
                 rtprintpanic!("{panicinfo}\nthread panicked while processing panic. aborting.\n");
             }
             panic_count::MustAbort::AlwaysAbort => {
```

Comment:
I could use the same logic as above in this branch as well, which would fix https://github.com/rust-lang/rust/issues/122940 but degrade panic printing for post-fork panics that do need formatting. Let me know what you think.

Reply:
@Amanieu any thoughts on this?

Reply:
I think https://github.com/rust-lang/rust/issues/122940 is better fixed with an additional flag on the global count.